### DISCUSSION TITLE
Handling token without MedusaJS Session

### DISCUSSION THREAD

**Discussion by markus-gx** [medusajs/medusa#7052] [Feature Requests]
When using different frontend frameworks, there are features that are different. In Nuxt 3 with SSR, for example, session management as currently provided by the medusa-js SDK is not really usable in terms of authentication and route middleware for protected routes. 
Hence the suggestion to manage the token itself (this makes it possible to handle server + client side hydration at the same time). Basically, everything is already there in the code base, except for a `setToken` function for the JWTTokenHandler. 
Adding this would make the SDK easier to customise for other frameworks.

**Comment by markus-gx** [comment]
PR: #7053

**Comment by chiuwah** [comment]
Just wondering could you elaborate how did you implement Medusa authentication in Nuxt 3 SSR and route middleware for protected routes?

**Reply by markus-gx** [reply]
> Just wondering could you elaborate how did you implement Medusa authentication in Nuxt 3 SSR and route middleware for protected routes?

Not yet. Thats why i made this PR. With the additional functionality i should be able to achieve that. Because currently the SDK is only working client side.

**Reply by chiuwah** [reply]
Would this be something for the Medusa Nuxt Module? [https://nuxt-medusa.vercel.app/](url)

**Reply by markus-gx** [reply]
> Would this be something for the Medusa Nuxt Module? [https://nuxt-medusa.vercel.app/](url)

Yeah sure. Using it already. Needs  to be modified as well afterwards. to handle authentication out of the box.
